src/core/services/interimElement/interimElement.js

```diff
@@ -421,9 +421,12 @@ function InterimElementProvider() {
          */
         function createAndTransitionIn() {
           return $q(function(resolve, reject){
-
             compileElement(options)
               .then(function( compiledData ) {
+                if (isRemoved) {
```

Comment:
what if `options.$destroy` is [set to false](https://github.com/angular/material/blob/fix/interim-remove-before-compile-finish/src/core/services/interimElement/interimElement.js#L463)?